fix: Out of order read rows fix

danieljbruce opened the pull request:
Client changes:

Emits an error if the chunk transformer receives keys that are out of order. The chunk transformer is a stream that data passes through before it reaches the stream that has the handlers for errors and retries. 

Also pulls some comparison functions into a utility file because now those functions are used by the chunk transformer to check to see if rows are in order. They are still used by the `table.ts` file, but are available in both places now.

Test proxy changes:

Makes it so that when the client emits an error as a result of rows that are out of order, that the error is sent back to the test runner as an rpc message instead of an rpc error. This is done to stay consistent with the test proxy in Java and allows the TestReadRows_NoRetry_OutOfOrderError test case to pass.

danieljbruce committed:
emit error if row keys are out of order

This commit emits an error if the chunk transformer recieves keys that are out of order. It also pulls some comparison functions into a utility file because now those functions are used by the chunk transformer too.

danieljbruce committed:
send the error from the client back as a result

This commit makes it so that when the client emits an error as a result of rows that are out of order, that the error is sent back to the test runner as an rpc message instead of an rpc error. This is done to stay consistent with the test proxy in Java and allows the TestReadRows_NoRetry_OutOfOrderError test case to pass.

danieljbruce committed:
Merge branch 'main' into out-of-order-read-rows-fix

ruyadorno commented on src/chunktransformer.ts:
```diff
@@ -259,6 +260,11 @@ export class ChunkTransformer extends Transform {
+      typeof lastRowKey !== 'undefined' &&
+      TableUtils.lessThanOrEqualTo(newRowKey as string, lastRowKey as string)
+    ) {
+      errorMessage = 'key must be strictly increasing';
```
What about a more friendly message, e.g: `A row key must be strictly increasing`

danieljbruce committed:
Change error message that gets raised for row key

The error message gets changed to match what is suggested in the pull request to be similar to other errors.
